### PR TITLE
Obsolete some spec-tests and examples

### PR DESCRIPTION
**Pull request by NikVolf**
There is non-exaustive list of obsolete spec tests

Since our current spec-tests harness is somewhat troublesome, we keep an absolute must minimum of it to be supported

And then migrate it to something more robust (it is yet to be defined)

At this PR, something that does not really test anything or shows something with examples is removed

upd: also removed test that tests nothing and just wastes cpu time

### CODE REVIEW

File: gclient/tests/perf.rs [hunk 1]
```diff
@@ -1,267 +0,0 @@
-use gclient::{GearApi, Result};
```
Comment (SabaunT) [inline gclient/tests/perf.rs:1]: This test had some important rationale, I guess. @shamilsan can you shed some light?

**Review by SabaunT — CHANGES_REQUESTED** [review]

File: gclient/tests/perf.rs (hunk 1 above)
Comment (NikVolf) [reply gclient/tests/perf.rs:1]: Well the reasons i mentioned when asked for that test was completely different as I recall, this is rather useless now

Comment (SabaunT) [reply gclient/tests/perf.rs:1]: discussed.

**Review by SabaunT — APPROVED** [review]

File: examples/Cargo.toml [hunk 2]
```diff
@@ -41,13 +37,10 @@ members = [
     "program-id",
     "rwlock",
     "rwlock_write",
-    "state_rollback",
-    "sum",
     "sync_duplicate",
-    "vec",
```
[the remark this answers is not in the document]
Comment (NikVolf) [reply examples/Cargo.toml:47]: This is covered by actual tests

The purpose of examples is to show that api is working, not allocator

[the remark this answers is not in the document]
Comment (grishasobol) [reply examples/Cargo.toml:47]: > This is covered by actual tests
Which test do you mean?
> The purpose of examples is to show that [gstd] api is working, not allocator
Yes, but we use them as tests in runtime-spec-tests and gear-test, so deleting them must be only with proof of already existing tests with same functionality or with replacing it to another one outside of our examples

[the remark this answers is not in the document]
Comment (NikVolf) [reply examples/Cargo.toml:47]: Even if there is no such test (which you should know better), it is not the purpose of examples / spec-tests to test allocator

They show API for users, and make sure that what is shown works

**Pull request merged**